### PR TITLE
chore: split up unit and integration tests, added tests

### LINKED ISSUE #667: Separation of Unit and Integration Tests for Fast Execution
**Description**
Currently, unit and integration tests are not strictly separated structurally. This forces developers to run the entire, time-consuming test suite even for minor local changes.

The goal of this ticket is to introduce a clear separation, creating a lightweight alternative for rapid local feedback during development.

**Problem**
- Long Feedback Loops: Because fast unit tests and heavier integration tests are coupled, there is no high-performance way to validate code changes locally within seconds.
- Barrier to Writing Tests: When executing the test suite takes too long, the incentive to thoroughly test edge cases and minor logic paths decreases. A more accessible test infrastructure is proven to improve testing discipline among developers.

**Objectives**
- Fast-Execution Shortcut: Provide a new command (e.g., make test-fast) that exclusively runs the lightweight tests.
- Performance Target: The execution time of this fast track should aim to be a maximum of ~5 seconds.

**Expected Benefit**
- Developer Experience: Significantly faster feedback loops during the implementation phase.
- Quality Improvement: Due to the improved and faster accessibility, overall test coverage is expected to increase, as writing edge-case tests becomes much easier.

### PR DESCRIPTION
**Pull request by rebEllieous**
## What
Split tests between unit- and integration-tests, added make targets for testing and for watching unit tests, added tests
Closes #667

## Why
Quicker feedback cycles for (some) tests, better developer ergonomy. 

## Testing
`make test` is unchanged
`make test-quick`: new target for unit tests
`make test-watch`: new target for watching unit tests -> takes ~5s at most

## Notes for reviewers
none

## Checklist
- [x] Tests added/updated
- [x] No breaking changes (or upgrade path documented above)
- [x] Readable commit history (squashed and cleaned up as desired)
- [x] AI code review considered and comments resolved

## Summary by CodeRabbit

* **New Features**
  * Added quicker Go test targets for running focused unit tests, including a watch mode for repeated local runs.

* **Bug Fixes**
  * Expanded validation coverage for Solar API resources to better catch missing or invalid fields.
  * Improved checks around updates so invalid changes are rejected more consistently.
  * Added coverage for controller and discovery behaviors, including retries, event handling, name truncation, and registry access grants.

* **Tests**
  * Added broad new test suites for Solar API, controller, and discovery components.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -85,6 +85,14 @@ test: $(SETUP_ENVTEST) $(GINKGO) envtest-binaries-sideload ocm-transfer-demo ##
 	 cat $(BUILD_PATH)/*_solar.full.coverprofile 2>/dev/null | grep -v '^mode:' >> $(BUILD_PATH)/solar.full.coverprofile || true
 	@grep -v 'zz_generated' $(BUILD_PATH)/solar.full.coverprofile > solar.coverprofile || true
 
+.PHONY: test-quick
```
Comment (yocaba) [inline Makefile]: Since "quick" says something about the speed but nothing about the type of tests, it might make sense to rename the targets to `make test-unit` and `make tests-unit-watch` e.g. That also aligns with the existing target `test-e2e`.

**Review by yocaba — APPROVED** [review]
lgtm
